src/Libraries/Microsoft.Extensions.AI.Abstractions/Embeddings/EmbeddingGeneratorExtensions.cs

```diff
@@ -18,7 +19,7 @@ public static class EmbeddingGeneratorExtensions
     /// <param name="options">The embedding generation options to configure the request.</param>
     /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
     /// <returns>The generated embedding for the specified <paramref name="value"/>.</returns>
-    public static Task<GeneratedEmbeddings<TEmbedding>> GenerateAsync<TValue, TEmbedding>(
+    public static async Task<TEmbedding> GenerateAsync<TValue, TEmbedding>(
```

Comment:
Don't we have a rule about naming methods differently when the return type changes?

Reply:
We try to avoid it. But unless we want to add Embeddings/Embedding to the interface/extension method names, respectively, I don't have a better option.